shared/js/ui/models/whitelist.es6.js

```diff
@@ -21,6 +22,31 @@ Whitelist.prototype = window.$.extend({},
         value: false
       }
       })
+    },
+
+    addDomain: function (url) {
+      // We only whitelist domains, not full URLs:
+      // - use getDomain, it will return null if the URL is invalid
+      // - prefix with getSubDomain, which returns an empty string if none is found
+      // But first, strip the 'www.' part, otherwise getSubDomain will include it
+      // and whitelisting won't work for that site
```

Comment:
Welp, this is a long comment, but I'd rather explain what's going on than leaving it to people's best guess